swarm.cabal

```diff
@@ -429,6 +429,56 @@ library swarm-web
     -- See discussion in #415
     StrictData
 
+library swarm-tournament
```

Comment:
Replying to the toplevel review comment here, so that the discussion can be properly threaded.

> I think the biggest issue is that we need to configure things so that by default, doing a `cabal build` does *not* build the tournament server, so it does not need to download and build servant, postgresql-simple, etc. Most people building `swarm` from source will not need all this so we should save them the trouble; anyone who actually wants the server can enable a flag.  I think we should be able to do this by creating a `flag` in the `.cabal` file and marking relevant libraries + executable stanzas as `buildable: false` conditional on the flag.  Or maybe there is a better way to accomplish this with modern cabal, I'm not sure.

Recall that the main `swarm` executable already includes `servant`, [from way back in 2022](https://github.com/swarm-game/swarm/pull/623/files#diff-42de5b60f981e95037ce731549b8f3d075c21bfa84407f76f56de17d5374c3deR150).

`postgresql-simple`, however, is a new dependency.  I think the simplest way to avoid building `postgresql-simple` is to pass an explicit target argument to cabal:
```
cabal build swarm:exe:swarm
```

As a convenience, we may provide a wrapper script (named `build-game.sh`) that makes the above invocation.

P.s. there seems to be a bug in `stack` that builds components/executables that are not requested on the command line.  So I think our documentation should explicitly recommend `cabal` in any places where it doesn't already.

Reply:
> Recall that the main `swarm` executable already includes `servant`, [from way back in 2022](https://github.com/swarm-game/swarm/pull/623/files#diff-42de5b60f981e95037ce731549b8f3d075c21bfa84407f76f56de17d5374c3deR150).

Ah, good point.  But avoiding `postgresql-simple` when not necessary is especially important since it has external library dependencies.  When I tried building I got an error about some thing that wasn't installed.  Getting it to work would have required me to go figure out which OS package to install that would provide whatever thing was missing.  But in any case, I think you're right that specifying a particular target like `swarm:exe:swarm` is the way to go, I had forgotten about that.

> P.s. there seems to be a bug in stack that builds components/executables that are not requested on the command line.

Really? It works for me.  e.g. when I do `stack build swarm:exe:swarm` it only builds the libraries needed for the main `swarm` executable.  By contrast just doing `stack build` builds other stuff like `swarm-scene` and `swarm-docs`.

Reply:
> Really? It works for me. e.g. when I do `stack build swarm:exe:swarm` it only builds the libraries needed for the main `swarm` executable. By contrast just doing `stack build` builds other stuff like `swarm-scene` and `swarm-docs`.

Try the following:
1. Add a newline to the end of `src/swarm-tournament/Swarm/Web/Tournament/Validate.hs`
2. Run `stack build swarm:exe:swarm`
3. Observe that it recompiles `swarm-tournament`, despite that library not being a dependency of the `swarm` executable.

Now do the same cycle, but with `cabal build swarm:exe:swarm` instead, and observe `Up to date` is printed.

Reply:
Hmm, you're right.

> I think our documentation should explicitly recommend `cabal` in any places where it doesn't already.

I think I'm convinced.  It just means anyone who wants to build from source has to pay more attention to what version of GHC they are using.  But these days with `ghcup` that's pretty trivial.

Reply:
Created #1820 to track this.